packages/babel-helper-split-export-declaration/src/index.js

```diff
@@ -0,0 +1,65 @@
+import * as t from "@babel/types";
+
+const splitExportDeclaration = exportDeclaration => {
+  if (!exportDeclaration.isExportDeclaration()) {
+    throw new Error("Only export declarations can be splitted.");
+  }
+
+  // build specifiers that point back to this export declaration
+  const isDefault = exportDeclaration.isExportDefaultDeclaration();
+  const declaration = exportDeclaration.get("declaration");
+
+  if (isDefault) {
+    if (declaration.isIdentifier()) {
+      throw new Error("It doesn't make sense to split exported identifier.");
+    }
+    const standaloneDeclaration =
+      declaration.isFunctionDeclaration() || declaration.isClassDeclaration();
+
+    let id = declaration.node.id;
+    const liveBinding = !!id;
+
+    if (!id) {
+      id = declaration.scope.generateUidIdentifier("default");
+      if (standaloneDeclaration) {
+        declaration.node.id = t.cloneNode(id);
+      }
+    }
+
+    const updatedDeclaration = standaloneDeclaration
+      ? declaration
+      : t.variableDeclaration("var", [
+          t.variableDeclarator(t.cloneNode(id), declaration.node),
+        ]);
+
+    const updatedExportDeclaration = liveBinding
+      ? t.exportNamedDeclaration(null, [
+          t.exportSpecifier(t.cloneNode(id), t.identifier("default")),
+        ])
+      : t.exportDefaultDeclaration(t.cloneNode(id));
+
+    exportDeclaration.insertAfter(updatedExportDeclaration);
+    exportDeclaration.replaceWith(updatedDeclaration);
+    return exportDeclaration;
+  }
+
+  if (exportDeclaration.get("specifiers").length > 0) {
+    throw new Error("It doesn't make sense to split exported specifiers.");
+  }
+
+  const bindingIdentifiers = declaration.getOuterBindingIdentifiers();
+
+  const specifiers = Object.keys(bindingIdentifiers).map(name => {
+    return t.exportSpecifier(t.identifier(name), t.identifier(name));
+  });
+
+  if (specifiers.length) {
```

Comment:
If there aren't specificers, should we throw an exception? Seems like there should always be some. If this is indented to handle `export { foo } from "bar";` we should probably bail out if there is a `source` instead.

Reply:
I've just extracted this functionality for reuse in other places, so I'm not sure what this if was supposed to handle.

I've added a `throw` in there and tests did pass, so I'm going to remove this `if` - this helper is intended to actually split things and is not supposed to do `maybeSplit*`, so I actually prefer not to swallow any cases here.

As to `export { foo } from "bar";` - this case is handled few lines above with:
```js
if (exportDeclaration.get("specifiers").length > 0) {
  throw new Error("It doesn't make sense to split exported specifiers.");
}
```

Reply:
> this case is handled few lines above with:

oh so it is, good call